src/app/static/src/js/components/runReport/runReport.vue

```diff
@@ -81,6 +82,32 @@
             }
         },
         methods: {
+            refreshGit: function () {
+                api.post('/git/fetch/')
+                    .then(() => {
+                        this.refreshPage()
+                    })
+                    .catch((error) => {
+                        this.error = error;
+                        this.defaultMessage = "An error occurred refreshing Git commits";
+                    });
+            },
+            refreshPage: function () {
+                if (this.metadata.git_supported) {
+                    this.selectedBranch = this.gitBranches[0];
+                    this.changedBranch();
+                } else {
+                    this.updateReports();
+                }
+                if (this.metadata.instances_supported) {
+                    const instances = this.metadata.instances;
+                    for (const key in instances) {
+                        if (instances[key].length > 0) {
+                            this.selectedInstances[key] = instances[key][0]
+                        }
+                    }
+                }
```

Comment:
This is independent of git, so running it more than once and anywhere but in `mounted` is unnecessary

Reply:
Agreed - I think rather than a monolithic refreshPage method, you could pull out sections which are run on `mounted` and on gitFetch into separate methods.